src/Microsoft.DotNet.Wpf/src/PresentationFramework/MS/Internal/WindowsRuntime/Generated/WinRT/ComWrappersSupport.cs

```diff
@@ -0,0 +1,187 @@
+                    Context.GetContextCallback());
+            }
+        }
+
+        public static List<ComInterfaceEntry> GetInterfaceTableEntries(object obj)
+        {
+            var entries = new List<ComInterfaceEntry>();
+            var interfaces = obj.GetType().GetInterfaces();
+            foreach (var iface in interfaces)
+            {
+                if (Projections.IsTypeWindowsRuntimeType(iface))
+                {
+                    var ifaceAbiType = iface.FindHelperType();
+                    entries.Add(new ComInterfaceEntry
+                    {
+                        IID = GuidGenerator.GetIID(ifaceAbiType),
+                        Vtable = (IntPtr)ifaceAbiType.FindVftblType().GetField("AbiToProjectionVftablePtr", BindingFlags.Public | BindingFlags.Static).GetValue(null)
+                    });
+                }
+
+                if (iface.IsConstructedGenericType
+                    && Projections.TryGetCompatibleWindowsRuntimeTypeForVariantType(iface, out var compatibleIface))
+                {
+                    var compatibleIfaceAbiType = compatibleIface.FindHelperType();
+                    entries.Add(new ComInterfaceEntry
+                    {
+                        IID = GuidGenerator.GetIID(compatibleIfaceAbiType),
+                        Vtable = (IntPtr)compatibleIfaceAbiType.FindVftblType().GetField("AbiToProjectionVftablePtr", BindingFlags.Public | BindingFlags.Static).GetValue(null)
+                    });
+                }
+            }
+
+            if (obj is Delegate)
+            {
+                entries.Add(new ComInterfaceEntry
+                {
+                    IID = GuidGenerator.GetIID(obj.GetType()),
+                    Vtable = (IntPtr)obj.GetType().GetHelperType().GetField("AbiToProjectionVftablePtr", BindingFlags.Public | BindingFlags.Static).GetValue(null)
+                });
+            }
+            // Add IAgileObject to all CCWs
+            entries.Add(new ComInterfaceEntry
+            {
+                IID = IID_IAgileObject,
+                Vtable = IUnknownVftbl.AbiToProjectionVftblPtr
+            });
+            return entries;
+        }
+
+        public static (InspectableInfo inspectableInfo, List<ComInterfaceEntry> interfaceTableEntries) PregenerateNativeTypeInformation(object obj)
+        {
+            var interfaceTableEntries = GetInterfaceTableEntries(obj);
+            var iids = new Guid[interfaceTableEntries.Count];
+            for (int i = 0; i < interfaceTableEntries.Count; i++)
+            {
+                iids[i] = interfaceTableEntries[i].IID;
+            }
+
+            Type type = obj.GetType();
+
+            if (type.FullName.StartsWith("ABI."))
+            {
+                type = Projections.FindCustomPublicTypeForAbiType(type) ?? type.Assembly.GetType(type.FullName.Substring("ABI.".Length)) ?? type;
+            }
+
+            return (
+                new InspectableInfo(type, iids),
```

Comment:
In fact, we did not use the type variable. So is it possible to remove the code that creates the type variable